packages/celotool/src/cmds/account/faucet.ts

```diff
@@ -40,18 +53,26 @@ export const handler = async (argv: FaucetArgv) => {
     console.log(`Using account: ${account}`)
     kit.defaultAccount = account
 
-    const [goldToken, stableToken] = await Promise.all([
+    const [goldToken, stableToken, reserve] = await Promise.all([
       kit.contracts.getGoldToken(),
       kit.contracts.getStableToken(),
+      kit.contracts.getReserve(),
     ])
-    const goldAmount = (await convertToContractDecimals(1, goldToken)).toString()
-    const stableTokenAmount = (await convertToContractDecimals(10, stableToken)).toString()
-
-    console.log(`Fauceting ${goldAmount} Gold and ${stableTokenAmount} StableToken to ${address}`)
-    await Promise.all([
-      goldToken.transfer(address, goldAmount).sendAndWaitForReceipt(),
-      stableToken.transfer(address, stableTokenAmount).sendAndWaitForReceipt(),
-    ])
+    const goldAmount = await convertToContractDecimals(argv.gold, goldToken)
+    const stableTokenAmount = await convertToContractDecimals(argv.dollars, stableToken)
+    console.log(
+      `Fauceting ${goldAmount.toFixed()} Gold and ${stableTokenAmount.toFixed()} StableToken to ${address}`
+    )
+    if (!goldAmount.isZero()) {
+      if (await reserve.isSpender(account)) {
+        await reserve.transferGold(address, goldAmount.toFixed()).sendAndWaitForReceipt()
+      } else {
+        await goldToken.transfer(address, goldAmount.toFixed()).sendAndWaitForReceipt()
+      }
+    }
+    if (stableTokenAmount.isZero()) {
```

Comment:
Probably future work, but without a separate faucet account we can't faucet C$ right now - but could if we spent from reserve then exchanged.